powersimdata/input/transform_profile.py

```diff
@@ -1,111 +1,98 @@
 import copy
 
-from powersimdata.input.grid import Grid
 from powersimdata.input.input_data import InputData
+from powersimdata.network.usa_tamu.constants.zones import interconnect2id
 
 
 class TransformProfile(object):
-    """Transforms profile according to operations listed in change table."""
+    """Transform profile according to operations listed in change table."""
 
     def __init__(self, scenario_info, grid, ct):
-        """Constructor
+        """Constructor.
 
         :param dict scenario_info: scenario information.
         :param powersimdata.input.grid.Grid grid: a Grid object.
         :param dict ct: change table.
         """
         self._input_data = InputData()
         self.scenario_info = scenario_info
-        self.grid = copy.deepcopy(grid)
+
         self.ct = copy.deepcopy(ct)
+        self.grid = copy.deepcopy(grid)
+
         self.scale_keys = {
-            "wind": {"wind", "wind_offshore"},
-            "solar": {"solar"},
-            "hydro": {"hydro"},
-            "demand": {"demand"},
+            "wind": ["wind", "wind_offshore"],
+            "solar": ["solar"],
+            "hydro": ["hydro"],
+            "demand": ["demand"],
         }
-        self.n_new_plant = (
-            0 if "new_plant" not in self.ct.keys() else len(self.ct["new_plant"])
-        )
-        self.base_plant = Grid(self.grid.interconnect).plant
+        self.n_new_plant, self.n_new_clean_plant = self._get_number_of_new_plant()
+
+    def _get_number_of_new_plant(self):
+        """Return the total number of new plant and new clean plant.
+
+        :return: (*tuple*) -- first element is the total number of new plant and second
+            element is the total number of new clean plant (*hydro*, *solar*,
+            *onshore wind* and *offshore wind*).
+        """
+        n_plant = [0, 0]
+        if "new_plant" in self.ct.keys():
+            for p in self.ct["new_plant"]:
+                n_plant[0] += 1
+                if p["type"] in sum(self.scale_keys.values(), []):
```

Comment:
Do we refactor `self.scale_keys` to be able to support this line? If so, we could do the same thing based on the current structure using something like `if p["type"] in set().union(*self.scale_keys.values()):`. I think they are equivalent.

Reply:
yeah, it was for the `sum`. Do you prefer to stick to `set`?

Reply:
I think a set is more correct when the order doesn't matter and there can be no duplicates, but it doesn't really matter here. Mostly I wanted to understand the meaning of the change.

Reply:
Done